keylime/tenant.py

```diff
@@ -639,7 +633,7 @@ def do_cv(self):
             "cloudagent_ip": self.cv_cloudagent_ip,
             "cloudagent_port": self.agent_port,
             "tpm_policy": json.dumps(self.tpm_policy),
-            "allowlist": json.dumps(self.allowlist),
```

Comment:
As Thore mentioned, this is an API change. Did you look at making it backwards compatible? If `allowlist` is provided treat it the same as before, if `ima_policy` is used do the new bundle processing? Not saying we have to have it that way, but if we're going to break backwards compat it should be for a good reason.

Reply:
Frankly, I didn't consider it, no. Though I think it would be good as a sanity check to lay out how the new bundle format works before introducing a backwards compatibility fix - I'd like to hear what you and Thore have to say about it, and make sure we think it's future-proof for the upcoming format shift for IMA policies.

I'll do that in a separate comment.

Reply:
I propose the following:
- Keep the `allowlist` option in the API. Treat all the data sent to it as before.
- Add the whole bundle feature to `ima_policy`, use it in the tenant only if a signature etc. is provided.
- Mark the `ima_policy` option as experimental until we have the new policy and bundle format in place.
- Once the new formats are implemented deprecate and remove `allowlist` from the API.

@mbestavros, @mpeters what do you think about that?

Reply:
@THS-on @mpeters I've implemented backwards compatibility on the CV, so nothing should functionally change if a legacy API request is made to an updated verifier anymore.

Given how much refactoring was done to allowlist reading and processing in this PR (and to the associated tests) to defer allowlist processing until the verifier, I'd prefer it if we didn't have to maintain legacy tenant functionality. It would add some complexity managing tests and functionality between the different types of inputs. But I can still look into it if that's preferred.

Reply:
I'm fine if we drop tenant verification of the signatures if it's now done at the verifier.